class undirected_exception(Exception):
    pass